src/header-utilities/exponentialSearch.hpp

```diff
@@ -0,0 +1,51 @@
+class ExponentialSearch {
+  template< typename RandomAccessIterator,
+	    typename Value,
+	    typename Compare >
+  static RandomAccessIterator apply( RandomAccessIterator left,
+				     RandomAccessIterator right,
+				     RandomAccessIterator end,
+				     Value&& value,
+				     Compare&& compare ){
+    if ( left == right ){ return right; }
+    const auto last = std::prev(right);
+    return ( not compare( *last, value ) ) ?
```

Comment:
The algorithm accepts a functor which defines some less than operation. What we're interested in is whether `value` is less than or equal to whatever the `last` iterator is pointing to. Because the comparison operation is user defined (which may be defined in terms of a subset of fields), we can't just do
` compare(value, *last) or (value == *last)`

However, we can get the less than or equal operation in terms of the less than operation. 
`(a <= b) == ( not ( b < a ) )`

You can see a table of these relationships [here](http://www.cplusplus.com/reference/list/list/operators/)